server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -2952,8 +2929,10 @@ public void run() {
         }
       }
     }
-    log.debug("Stopping Replication Server");
-    TServerUtils.stopTServer(this.replServer);
+    if (this.replServer != null) {
```

Comment:
If `replServer` is not volatile then may not see changes from another thread.

Reply:
Fixed in 0016dce1a80f7c8f8552252f148ba77f94da2725